File: common/src/main/java/net/impleri/playerskills/integration/ftbquests/SpecializedSkillTask.java
package net.impleri.playerskills.integration.ftbquests;

import dev.ftb.mods.ftbquests.quest.Quest;
import dev.ftb.mods.ftbquests.quest.task.TaskType;
import net.impleri.playerskills.variant.specialized.SpecializedSkillType;
import net.minecraft.resources.ResourceLocation;

public class SpecializedSkillTask extends TieredSkillTask {
    public SpecializedSkillTask(Quest quest) {
        super(quest);
    }

    @Override
    protected ResourceLocation getSkillType() {
        return SpecializedSkillType.name;
    }

    @Override
    public TaskType getType() {
        return SkillTaskTypes.SPECIALIZED_SKILL;
    }
}

File: common/src/main/kotlin/net/impleri/playerskills/integrations/ftbquests/SpecializedSkillTask.kt
package net.impleri.playerskills.integrations.ftbquests

import dev.ftb.mods.ftbquests.quest.Quest
import dev.ftb.mods.ftbquests.quest.task.TaskType
import net.impleri.playerskills.skills.specialized.SpecializedSkillType
import net.minecraft.resources.ResourceLocation

class SpecializedSkillTask(quest: Quest?) : TieredSkillTask(quest) {
  override val skillType: ResourceLocation
    get() = SpecializedSkillType.NAME

  override fun getType(): TaskType {
    return SkillTaskTypes.SPECIALIZED_SKILL
  }
}
